src/codegen/compile.ml

```diff
@@ -3057,8 +3392,9 @@ module MakeCompact (Num : BigNumType) : BigNumType = struct
           get_amount ^^ compile_rel_const I32Op.LeU 42l ^^
           G.i (Binary (Wasm.Values.I32 I32Op.And)) ^^
           G.if1 I32Type
-            get_res
-            (get_n ^^ compile_shrS_const 1l ^^ Num.from_word30 env ^^ get_amount ^^ Num.compile_lsh env)
+            (get_res ^^ compile_bitor_const (TaggingScheme.tag_of_typ Type.Int))
+            (get_n ^^ compile_shrS_const (Int32.of_int (32 - BitTagged.ubits_of Type.Int)) ^^
+             Num.from_word30 env ^^ get_amount ^^ Num.compile_lsh env) (* TBR: from_word30? *)
```

Comment:
I am not sure whether I understand `from_word30`:
* In the original version, it maybe represented the 30 bits payload of the unboxed signed integer (32 minus 1 bit for the old tag and minus the sign bit, even if it was Nat).
* In the new version of precise tagging, it would probably be 29 bits (as the tag now occupies two bits), as there is no distinguishing between signed and unsigned big numbers? 
Maybe we could rename `from_word30` to e.g. `tag_small_bignum`?